canonical-webteam/deployment-configs
# New structure for config files

One or two changes...:

- Add namespaces - staging and production
- Document in README
- Use a single config file for each set of service+deployment configs for a domain, which we then apply to staging and production namespaces separately
- Keep service configs in `services`
- Keep ingress configs in `ingresses/production` and `ingresses/staging`, one file per domain
- Add `X-Robots-Tag` headers to staging domains
- Add `imagePullPolicy: Always` to all deployment configs
  This is so that whenever a `deploy-to-{staging|production}` job is run it will re-pull the image. This means that if there are no *code* changes (so the image tag will remain the same, as the Git commit ID), but there have instead been security patches added to `ubuntu:xenial`, we can still apply these security patches through a simple deployment.

QA
--

This can be QAed with [minikube](https://kubernetes.io/docs/getting-started-guides/minikube/) - e.g. see https://github.com/canonical-webteam/k8s-deployments/pull/10
